### ISSUE TITLE
[Bug Report] Paths should be relative to config file, NOT current working directory of the executable

### ISSUE THREAD

**Issue by aghoulcoder** [stashapp/stash#3029]
Currently, relative paths (for `generated/`, `plugins/`, and `stash-go.sqlite`) are relative to the current working dir (cwd) of the executable. *This serves no useful purpose*. It also has the potential to cause unexpected behavior and even data loss.

I currently think that the simplest acceptable solution is to make relative paths relative to the config file.

I plan to work on a patch for this which is why I'm skipping a more detailed description and reproducer. But I still wanted to open an issue in case somebody else is thinking about this issue and wants to share an opinion or suggestion.

**Comment by HijackHornet** [comment]
I agree. I'm using the same exe for 2 instances in separate folder. 
To avoid issue i ended up writing fullpaths to each plugins, scraper etc folder.
But i think cwd is standard

**Comment by aghoulcoder** [comment]
As I'm looking into this it appears this is actually *a proper bug*. During initia
l setup, if you leave the paths blank, it tells you that it will default to `<path containing configuration file>`. But then the app actually tries to write to the `cwd`!
![bug](https://user-images.githubusercontent.com/115620677/197292590-788cf877-90a8-487b-8c14-e3848e2da4d3.png)

**Comment by aghoulcoder** [comment]
Also, I would propose using [XDG Base Directory Specification](https://golangexample.com/go-implementation-of-the-xdg-base-directory-specification-and-xdg-user-directories/) for determining *default* config and data paths. Currently they are "hardcoded". 

One notable offender is `~/.stash`. There is a general effort in Linux to stop polluting the user's home dir with dotfiles/dotdirs. In this case `~/.stash` should be split between config and data. The config file should go into `~/.config/stash/config.yml`. The data dirs should go under `~/.local/share/stash/`.

But even then, these paths *must not* be hardcoded. The `~/.config` and `~/.local/share` examples are just the XDG defaults. Instead, the application (stash) should follow the XDG *environment variables* if they are set, then fall-back to the XDG defaults. Most of this appears to be handled by the [adrg/xdg](https://github.com/adrg/xdg) package I linked to above.

Apparently (and I just learned this now) XDG also defines paths for MacOS and Windows, not just Linux. So it's a comprehensive cross-platform standard.
